docs/versioned_docs/version-0.13.1/general/04-Executing Automations/multi-device-automation.md

```diff
@@ -2,193 +2,349 @@
+### 2. Start AskUI Controller for Each Device
+You __must__ start a separate AskUI Controller for each Android device you are automating. The `-p` argument sets the port the AskUI Controller is listening on for commands. Choose a different one for each device.
+
+:::tip
+
+The `-d` flag specifies to which device a AskUI Controller instance connects.
+
+Run `adb devices` in a terminal to get a list of connected devices. In the following example, if you want to connect to `emulator-1` you use `-d 0` and for `emulator-2` you use `-d 1`.
 
 ```bash
-[2023-01-02 17:31:19.634 +0100] DEBUG (AskuiUiController): Window is minimized.
-[2023-01-02 17:31:19.639 +0100] INFO (AskuiUiController): Selecting display number 0.
-[2023-01-02 17:31:19.641 +0100] INFO (AskuiUiController): Successfully started.
+$ adb devices
+emulator-1 device product:sdk_google_phone_x86_64 model:Android_SDK_built_for_x86_64 device:generic_x86_64
+emulator-2 device product:sdk_google_phone_x86 model:Android_SDK_built_for_x86 device:generic_x86
 ```
 
-### Android Only:
-If your remote device is an Android device, run the *AskUI Controller* on the **local device (desktop)** with an extra option as shown below:
+:::
+
+<Tabs>
+  <TabItem value="windows" label="Windows" default>
+
+Set a different port for each AskUI Controller and specify which Android device to connect to with the `-d` flag.
 
 ```bash
-# Windows PowerShell
-.\askui-ui-controller.exe --host 0.0.0.0 -d 0 -m -r android
+# Activate AskUI Development Environment (ADE) first
+# Connects to the first device returned by 'adb devices'
+AskUI-RunController --host 0.0.0.0 -p 6769 -d 0 -m
 
-# Windows cmd
-start askui-ui-controller.exe --host 0.0.0.0 -d 0 -m -r android
+# Connects to the second device returned by 'adb devices'
+AskUI-RunController --host 0.0.0.0 -p 6770 -d 1 -m
+```
 
-# macOS
-./askui-ui-controller --host 0.0.0.0 -d 0 -m -r android
+  </TabItem>
+  <TabItem value="linux" label="Linux" default>
 
-# Linux
-./askui-ui-controller.AppImage --host 0.0.0.0 -d 0 -m -r android
+Set a different port for each AskUI Controller and specify which Android device to connect to with the `-d` flag.
+
+```bash
+# Connects to the first device returned by 'adb devices'
+./askui-ui-controller.AppImage --host -p 6769 0.0.0.0 -d 0 -m -r android
+
+# Connects to the second device returned by 'adb devices'
+./askui-ui-controller.AppImage --host -p 6770 0.0.0.0 -d 1 -m -r android
 ```
 
-Make sure that your local device (desktop) is running **TWO DIFFERENT `askui-ui-controller`**, if you want to control the local device and the Android device at the same time.
+  </TabItem>
+  <TabItem value="macos" label="macOS" default>    
 
-## 4. Write the AskUI Code
+Set a different port for each AskUI Controller and specify which Android device to connect to with the `-d` flag.
+
+```bash
+# Connects to the first device returned by 'adb devices'
+./askui-ui-controller --host 0.0.0.0 -p 6769 -d 0 -m -r android
```

Comment:
The `askui-ui-controller` is blocking the shell on mac & linux. We need to start it in two different shells. Or start it in background.

Reply:
I prefer the foreground because of the logs. Will add an explanation of what to do!